api/services/export-data-2.js

```diff
@@ -99,31 +100,165 @@ const CSV_MAPPER = {
         return {
           header: allColumns,
           fn: record => {
-            return [
+            return [[
               record._id,
               record.form,
               record.patient_id,
-              record.reported_date,
+              formatDate(record.reported_date),
               record.from,
               objectPath.get(record, ['contact', 'name']),
               objectPath.get(record, ['contact', 'parent', 'name']),
               objectPath.get(record, ['contact', 'parent', 'parent', 'name']),
               objectPath.get(record, ['contact', 'parent', 'parent', 'parent', 'name'])
-            ].concat(fieldColumns.map(c => objectPath.get(record.fields, c)));
+            ].concat(fieldColumns.map(c => objectPath.get(record.fields, c)))];
           }
         };
       }));
   },
 
-
-  // NB: we don't actually support this (note SUPPORTED_EXPORTS above), it's
-  // here in the code to show a simpler examplen than the complicated forms one
   contacts: () => Promise.resolve({
     header: ['id', 'rev', 'name', 'patient_id', 'type'],
-    fn: record => [record._id, record._rev, record.name, record.patient_id, record.type]
+    fn: record => [[record._id, record._rev, record.name, record.patient_id, record.type]]
+  }),
+
+  messages: () => Promise.resolve({
+    header: [
+      'id',
+      'patient_id',
+      'reported_date',
+      'from',
+      'type',
+      'state',
+      'received',
+      'scheduled',
+      'pending',
+      'sent',
+      'cleared',
+      'muted',
+      'message_id',
+      'sent_by',
+      'to_phone',
+      'content'
+    ],
+    fn: record => {
+      const tasks = normalizeTasks(record);
+      return _.flatten(tasks.map(task => {
+        const history = buildHistory(task);
+        return task.messages.map(message => [
+          record._id,
+          record.patient_id,
+          formatDate(record.reported_date),
+          record.from,
+          task.type || 'Task Message',
+          task.state,
+          getStateDate('received', task, history),
+          getStateDate('scheduled', task, history),
+          getStateDate('pending', task, history),
+          getStateDate('sent', task, history),
+          getStateDate('cleared', task, history),
+          getStateDate('muted', task, history),
+          message.uuid,
+          message.sent_by,
+          message.to,
+          message.message
+        ]);
+      }), true);
+    }
   })
 };
 
+const normalizeResponse = doc => {
```

Comment:
I'm not a fan of this function and their friends being scoped to the whole file for use in one export, it kind of makes this file a dumping ground. In the existing export we were able to keep these inline, but perhaps this is a sign that each export needs to be split out into its own file. Could even use classes if you want to align the API clearly (though since there are no constrains AFAIK it's no different really from just `module.exports`